lang/csharp/src/apache/test/Interop/InteropDataTests.cs

```diff
@@ -0,0 +1,52 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+using System.IO;
+using NUnit.Framework;
+using Avro.File;
+using Avro.Generic;
+
+namespace Avro.Test.Interop
+{
+    [TestFixture]
+    public class InteropDataTests
+    {
+        [TestCase("../../../../../../../../build/interop/data")]
+        public void TestInterOp(string inputDir)
+        {
```

Comment:
The "current directory" varies depending on how you run this test (e.g. `dotnet test` vs running tests in Visual Studio's Test Explorer). It also varies in .NET Framework vs .NET Core. For example, running on my Windows machine, this tests passes in .NET Core but fails in .NET Framework because the current directory is different.

I was able to resolve this by resolving the `inputDir` relative to the `TestDirectory` rather than the "current directory".

```suggestion
        {
            // Resolve inputDir relative to the TestDirectory
            inputDir = Path.Combine(TestContext.CurrentContext.TestDirectory, inputDir);

            Assert.True(Directory.Exists(inputDir),
                "Input directory does not exist. Run `build.sh interop-data-generate` first.");

```

The next `Assert.True` ensures that we fail with a helpful message, in the event that a developer runs all tests without first generating the interop data. The failure message indicates the manual steps required to make the test pass.